### PR TITLE
fix(taskframework): oss log info is ignored when cancel job

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-bug
module-taskframework
#### What this PR does / why we need it:
log info is  missing when cancel job or close task-framework, so we should update oss `LOG_STORAGE_ALL_OBJECT_ID` and `LOG_STORAGE_BUCKET_NAME` in job_attribute in report result

#### Special notes for your reviewer:

- case 1: 

1. start job, job is running and can show log
2. cancel job 
result: log can be show normally

- case 2:

1. start job, job is running and can show log
2. close task-framework and restart
result: job is failed and log can be show normally

above 2 cases has test passed
#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 1]
```diff
@@ -310,8 +313,11 @@ public void handleResult(TaskResult taskResult) {
             log.warn("Job identity is not exists by id {}", taskResult.getJobIdentity().getId());
             return;
         }
-        if (je.getStatus().isTerminated() || je.getStatus() == JobStatus.CANCELING) {
-            log.warn("Job {} is finished, ignore result, currentStatus={}", je.getId(), je.getStatus());
+        if (je.getStatus() == JobStatus.CANCELING || !taskFrameworkEnabledProperties.isEnabled()) {
+            saveOrUpdateLogMetadata(taskResult, je.getId(), je.getStatus());
+            return;
+        } else if (je.getStatus().isTerminated()) {
+            log.warn("Job is finished, ignore result, jobId={}, currentStatus={}", je.getId(), je.getStatus());
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:320]: why ignore result while job is finieshed ?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 2]
```diff
@@ -310,8 +313,11 @@ public void handleResult(TaskResult taskResult) {
             log.warn("Job identity is not exists by id {}", taskResult.getJobIdentity().getId());
             return;
         }
-        if (je.getStatus().isTerminated() || je.getStatus() == JobStatus.CANCELING) {
-            log.warn("Job {} is finished, ignore result, currentStatus={}", je.getId(), je.getStatus());
+        if (je.getStatus() == JobStatus.CANCELING || !taskFrameworkEnabledProperties.isEnabled()) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:316]: why save log meta while  task framework not enabled

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:316]: task is running before close task framework,   when close task framework we send stop command to pod, then task will be stopped and upload log to oss,  we save the report result  about oss info

**Review by yizhouxw — APPROVED** [review]
merge first, may refactor later

**Pull request merged**